src/Mod/Assembly/App/AssemblyObject.cpp

```diff
@@ -2185,10 +2187,177 @@ App::DocumentObject* AssemblyObject::getObjFromProp(App::DocumentObject* joint,
     return propObj->getValue();
 }
 
-App::DocumentObject* AssemblyObject::getLinkedObjFromProp(App::DocumentObject* joint,
-                                                          const char* pObj)
+App::DocumentObject* AssemblyObject::getObjFromRef(App::PropertyXLinkSub* prop)
+{
+    if (!prop) {
+        return nullptr;
+    }
+
+    App::Document* doc = prop->getValue()->getDocument();
+
+    std::vector<std::string> names = getSubAsList(prop);
+
+    // Lambda function to check if the typeId is a BodySubObject
+    auto isBodySubObject = [](App::DocumentObject* obj) -> bool {
+        // getViewProviderName instead of isDerivedFrom to avoid dependency on sketcher
+        return (obj->getViewProviderName() == "SketcherGui::ViewProviderSketch"
```

Comment:
This is a char* to char* comparison, you can't use operator==.

Reply:
Thanks for the catch !